src/classes/dbus-handler/dbus-handler.cpp

```diff
@@ -0,0 +1,201 @@
+#include "dbus-handler.hpp"
+
+auto DBusHandler::findObject(const DBusHandler::Path& path) -> sdbus::IObject*
+{
+    try {
+        return m_DBusObjects.at(path.objectPath).get();
+    } catch (std::out_of_range& e) {
+        m_DBusObjects[path.objectPath] = sdbus::createObject(*m_connection, path.objectPath);
+        return m_DBusObjects[path.objectPath].get();
+    }
+}
+
+auto DBusHandler::findProxy(const DBusHandler::Path& path) -> sdbus::IProxy*
+{
+    const std::string uniqueId = path.service + path.objectPath;
+
+    try {
+        return m_DBusProxys.at(uniqueId).get();
+    } catch (std::out_of_range& e) {
+        m_DBusProxys[uniqueId] = sdbus::createProxy(path.service, path.objectPath);
+        return m_DBusProxys[uniqueId].get();
+    }
+}
+
+DBusHandler::DBusHandler(const std::string& serviceName)
+    : m_isServer { true }
+    , m_serviceName { serviceName }
+{
+    m_connection = sdbus::createSystemBusConnection(serviceName);
+}
+
+DBusHandler::DBusHandler()
+    : m_isServer { false } {};
+
+void DBusHandler::registerMethod(const DBusHandler::Path& path, DBusCallback&& callback)
```

Comment:
N seria legal liberar lambdas também em lvalue?
Daria pra fazer isso assim
```suggestion
void DBusHandler::registerMethod(const DBusHandler::Path& path, const DBusCallback& callback)
```

Reply:
Não funciona com lvalue nesse caso, já que esse recurso será reutilizado.

Reply:
Acho que entendi, blz então